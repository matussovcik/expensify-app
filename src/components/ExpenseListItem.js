import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <h3>{description}</h3>
    <Link to={`/edit/${id}`}>Edit</Link>
    <p>Amount: {amount}</p>
    <p>Created at: {createdAt}</p>
  </div>
);

export default ExpenseListItem;