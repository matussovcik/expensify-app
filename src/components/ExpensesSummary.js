import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensesTotal';

export const ExpensesSummary = (props) => (
  <div>
    <p>Viewing {props.expenses.length} {props.expenses.length === 1 ? 'expense' : 'expenses'} totalling: {numeral(selectExpensesTotal(props.expenses) / 100).format('$0,0.00')}.</p>
  </div>
);

const mapStateToProps = (state) => ({
  expenses: selectExpenses(state.expenses, state.filters)
});

export default connect(mapStateToProps)(ExpensesSummary);