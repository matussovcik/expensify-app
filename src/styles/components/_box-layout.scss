.box-layout {
  background: url('/images/bg.jpg');
  height: 100vh;
  width: 100vw;
  display: flex;
  background-size: cover;
  background-repeat: no-repeat;
  justify-content: center;
  align-items: center;

  &__box {
    background: fade-out(white, .15);
    border-radius: 3px;
    text-align: center;
    width: 25rem;
    padding: $l-size $m-size;
  }

  &__title {
    margin: 0 0 $m-size 0;
    line-height: 1;
  }
}