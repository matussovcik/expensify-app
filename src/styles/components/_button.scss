.btn {
  border: none;
  cursor: pointer;
  display: inline-block;
  padding: $s-size;
  background-color: $blue;
  font-size: $font-size-large;
  font-weight: $thin;
  color: white;
  text-decoration: none;
  line-height: 1;

  &--link {
    background: none;
  }

  &--grey {
    background-color: #888;
  }
}