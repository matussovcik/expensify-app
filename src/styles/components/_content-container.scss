.content-container {
  max-width: 80rem;
  padding: 0 $m-size;
  margin: 0 auto;
}

.content-justify {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: $s-size 0;
}