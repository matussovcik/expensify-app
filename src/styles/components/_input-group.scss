.input-group {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  align-items: center;
  margin-bottom: $m-size;

  @media only screen and (min-width: $desktop-breakpoint) {
    flex-direction: row;
    margin-bottom: $l-size;
  }

  &__item {
    margin-bottom: $s-size;
    
    @media only screen and (min-width: $desktop-breakpoint) {
      margin-right: $s-size;
      margin-bottom: 0;
    }
  }
}