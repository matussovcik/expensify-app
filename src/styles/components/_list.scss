.list {
  &__header {
    background-color: $off-white;
    border: 1px solid darken($off-white, 7%);
    color: $grey;
    display: flex;
    justify-content: space-between;
    padding: $s-size $m-size;
  }

  &__item {
    border: 1px solid darken($off-white, 7%);
    border-top: none;
    color: $dark-grey;
    display: flex;
    flex-direction: column;
    padding: $s-size;
    text-decoration: none;
    transition: background-color .3s ease;

    @media only screen and (min-width: $desktop-breakpoint) {
      flex-direction: row;
      justify-content: space-between;
      align-items: center;
      padding: $m-size;
    }

    &:hover {
      background-color: $off-white; 
    } 

    &--message {
      align-items: center;
      justify-content: center;
      color: $grey;

      &:hover {
        background: none; 
      } 
    }
  }

  &__item-title {
    margin: 0;
    word-break: break-all;
  }

  &__item-subtitle {
    color: $grey;
    font-size: $font-size-small;
  }

  &__item-data {
    margin-top: $s-size;

    @media only screen and (min-width: $desktop-breakpoint) {
      margin-top: 0;
      padding: $s-size;
    }
  }

  &__body {
    margin-bottom: $m-size;

    @media only screen and (min-width: $desktop-breakpoint) {
      margin-bottom: $l-size;
    }
  }
}