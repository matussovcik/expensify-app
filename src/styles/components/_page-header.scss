.page-header {
  background-color: $off-white;
  padding: $l-size 0;
  margin-bottom: $l-size;

  &__actions {
    margin-top: $m-size;
  }

  &__title {
    margin: 0;
    font-weight: $thin;

    span {
      font-weight: $bold;
    }
  }
}

  