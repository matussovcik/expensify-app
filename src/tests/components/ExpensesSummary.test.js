import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';

test('should get correct number of expenses viewed and the total amount', () => {
  const wrapper = shallow(<ExpensesSummary expenses={expenses}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should get 1 expense and its amount', () => {
  const wrapper = shallow(<ExpensesSummary expenses={[expenses[0]]}/>);
  expect(wrapper).toMatchSnapshot();
});