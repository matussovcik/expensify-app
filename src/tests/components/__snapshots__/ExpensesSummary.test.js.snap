// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should get 1 expense and its amount 1`] = `
<div>
  <p>
    Viewing 
    1
     
    expense
     totalling: 
    $1.93
    .
  </p>
</div>
`;

exports[`should get correct number of expenses viewed and the total amount 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses
     totalling: 
    $191,752.17
    .
  </p>
</div>
`;
